src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs

```diff
@@ -24,24 +24,22 @@ public class ObjectCollection : IList
 
             public ObjectCollection(ListBox owner)
             {
-                this.owner = owner;
+                this.owner = owner ?? throw new ArgumentNullException(nameof(owner));
             }
 
             /// <summary>
-                    ///  Initializes a new instance of ListBox.ObjectCollection based on another ListBox.ObjectCollection.
-                /// </summary>
-            public ObjectCollection(ListBox owner, ObjectCollection value)
+            ///  Initializes a new instance of ListBox.ObjectCollection based on another ListBox.ObjectCollection.
+            /// </summary>
+            public ObjectCollection(ListBox owner, ObjectCollection value) : this(owner)
             {
-                this.owner = owner;
                 AddRange(value);
             }
 
             /// <summary>
-                    ///  Initializes a new instance of ListBox.ObjectCollection containing any array of objects.
-                /// </summary>
-            public ObjectCollection(ListBox owner, object[] value)
+            ///  Initializes a new instance of ListBox.ObjectCollection containing any array of objects.
+            /// </summary>
+            public ObjectCollection(ListBox owner, object[] value) : this(owner)
             {
```

Comment:
```suggestion
            public ObjectCollection(ListBox owner, object[] value) : this(owner)
            {
                value ??= throw new ArgumentNullException(nameof(value));
```

I think it is better to hoist the exception up rather get it from calling `AddRange(value)`.


In fact looking through the API, I don't think `AddRangeInternal` must be `internal` or validate `value` argument. The call sites in `ListBox` should be replaced with `AddRange`, as there is nothing that requires `AddRangeInternal` there... This isn't a direction for action, just sharing my thoughts here.

Reply:
Yes, I agree this makes sense.

Also has the added benefit of actually making the `paramName` in `ArgumentNullException` actually match the parameter name in the method/constructor